lib/+mwbs/+Visualizers/+robotVisualizer/iDynTreeIrrlichtVisualizer.m

```diff
@@ -0,0 +1,102 @@
+classdef iDynTreeIrrlichtVisualizer < matlab.System
+    % RobotVisualizer matlab.System handling the robot visualization. Based
+    % on iDynTree-Irrlicht-Visualizer bindings.
+    % go in app/robots/iCub*/initRobotVisualizer.m to change the setup config
+    
+    properties (Nontunable)
+        config;
+        min_time_viz = 1/25;
+    end
+    
+    properties (DiscreteState)
+        
+    end
+    
+    % Pre-computed constants
+    properties (Access = private)
+        KinDynModel, viz;
+        g = [0, 0, -9.81]; % gravity vector
+    end
+    
+    methods (Access = protected)
+        
+        function setupImpl(obj)
+            
+            if obj.config.visualizeRobot
+                % Perform one-time calculations, such as computing constants
+                obj.prepareRobot()
+                tic;
+            end
+            
+        end
+        
+        function icon = getIconImpl(~)
+            % Define icon for System block
+            icon = ["iDynTree-Irrlicht", "Visualizer"];
+        end
+        
+        function stepImpl(obj, world_H_base, base_velocity, joints_positions, joints_velocity)
+            % stepImpl Specifies the algorithm to execute when you run the System object
+            if obj.config.visualizeRobot
+                % take the kinematic quantities and set the robot state
+                iDynTreeWrappers.setRobotState(obj.KinDynModel, world_H_base, joints_positions, base_velocity, joints_velocity, obj.g);
+                if obj.viz.run()
+                    time_interval = toc;
+                    % check the maximum fps
+                    if time_interval > obj.min_time_viz
```

Comment:
It regulates the maximum frame-per-seconds, in this way the visualizer refresh rate is bounded and we don't waste resources. 
To measure the time passed between one refresh and the following one I choose `tic` `toc` function. Honestly, I don't know if it is the less demanding strategy, although I think is the cleanest.

Reply:
Yes got it. Actually my doubt was about `obj.viz.draw()`, wether it is blocking or not.

For `minSampleTime` to be actually the maximum FPS, you should call `tic` before `obj.viz.draw()` (still in the "if" block of course). This way you include the processing time of `obj.viz.draw()` in case this call is blocking.

**Side note:** But this doesn't prevent the `draw` to take too much resources if its processing time gets close to `minSampleTime`.

## limiting the resources used by the visualizer
The present code with tic...toc makes sure to leave `minSampleTime` time for all the processing outside `obj.viz.draw()` between two consecutive calls to that function.

This is ok for starting. If there's a way to improve it, we can do it in another PR.